src/ngraph_graph.h

```diff
@@ -296,6 +310,7 @@ class Graph : public Node {
   std::vector<std::shared_ptr<OutputElement>> output_elements_;
   std::vector<bool> input_is_weight_;
   bool zero_grad = false;
+  bool need_grad = true;
```

Comment:
Could you please add a comment why need_grad is true by default and is handled differently in one specific Compiler() constructor?